### PR TITLE
Add Xcode Command Line Tools version to Software report

### PR DESCRIPTION
**Pull request by dsame**
# Description
Improvement

I git the idea from brew code and use clang version as a key to discover the version of Xcode the CLT came from. 
The output looks like:

```
2020-12-08T01:38:31.9591921Z #### Unit Test Framework
2020-12-08T01:38:31.9592616Z - NUnit 3.6.1
2020-12-08T01:38:31.9593022Z 
2020-12-08T01:38:31.9593377Z ### Xcode
2020-12-08T01:38:31.9593848Z | Version          | Build   | Path                           |
2020-12-08T01:38:31.9594703Z | ---------------- | ------- | ------------------------------ |
2020-12-08T01:38:31.9595374Z | 11.3.1 (default) | 11C505  | /Applications/Xcode_11.3.1.app |
2020-12-08T01:38:31.9596059Z | 11.3             | 11C29   | /Applications/Xcode_11.3.app   |
2020-12-08T01:38:31.9596705Z | 11.2.1           | 11B500  | /Applications/Xcode_11.2.1.app |
2020-12-08T01:38:31.9597342Z | 11.2             | 11B52   | /Applications/Xcode_11.2.app   |
2020-12-08T01:38:31.9597942Z | 11.1             | 11A1027 | /Applications/Xcode_11.1.app   |
2020-12-08T01:38:31.9598506Z | 11.0             | 11A420a | /Applications/Xcode_11.app     |
2020-12-08T01:38:31.9599229Z | 10.3             | 10G8    | /Applications/Xcode_10.3.app   |
2020-12-08T01:38:31.9599803Z | 10.2.1           | 10E1001 | /Applications/Xcode_10.2.1.app |
2020-12-08T01:38:31.9600356Z | 10.2             | 10E125  | /Applications/Xcode_10.2.app   |
2020-12-08T01:38:31.9600906Z | 10.1             | 10B61   | /Applications/Xcode_10.1.app   |
2020-12-08T01:38:31.9601459Z | 10.0             | 10A255  | /Applications/Xcode_10.app     |
2020-12-08T01:38:31.9602102Z | 9.4.1            | 9F2000  | /Applications/Xcode_9.4.1.app  |
2020-12-08T01:38:31.9602576Z 
2020-12-08T01:38:31.9603196Z - Command Line Tools: 10.3
```

## Check list
- [x] Related issue / work item is attached
- [x] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images/macos/software-report/SoftwareReport.Xcode.psm1 [hunk 1]
```diff
@@ -220,6 +220,40 @@ function Build-XcodeSimulatorsTable {
...
 
+function Build-XcodeCommandLineToolsSection {
+    $PKG_PATH = "/Library/Developer/CommandLineTools"
+
+    $clang_version = (&"${PKG_PATH}/usr/bin/clang" --version).Split(' ')[3]
+
+    $clang_version_parts = $clang_version.Split('.')
+    $clang_version_no = ($clang_version_parts[0] -as [int]) * 10 + ($clang_version_parts[1] -as [int])
+
+    $xcode_version = switch ($clang_version_no) {
```
Comment (maxim-lobanov) [inline images/macos/software-report/SoftwareReport.Xcode.psm1:231]: Is there any way to avoid maintenance of this list? Looks like we will have to update it manually in future.
I would even say that this list is already outdated for now.

I think we can retrieve CLT version via pkgutil tool: `pkgutil --pkg-info com.apple.pkg.CLTools_Executables`
```
pkgutil --pkg-info com.apple.pkg.CLTools_Executables | grep "version" | cut -d' ' -f 2 | cut -d'.' -f 1,2
```
should work without maitenance

File: images/macos/software-report/SoftwareReport.Xcode.psm1 [hunk 2]
```diff
@@ -220,6 +220,40 @@ function Build-XcodeSimulatorsTable {
...
+        90 {"9.2"}
+        91 {"9.4"}
+        100 {"10.3"}
+        110 {"11.5"}
+        default   {"12.0"}
+    }
+
+    $versions = @(
+        "clang version $clang_version",
+        "Corresponding XCode version $xcode_version"
```
Comment (maxim-lobanov) [inline images/macos/software-report/SoftwareReport.Xcode.psm1:248]: What does it mean `Corresponding XCode version`?
Command Line Tools is not related to Xcode. They have similar versioning but it is a separate tool - replacement of Xcode.

I suggest the following

File: images/macos/software-report/SoftwareReport.Generator.ps1 [hunk 3]
```diff
@@ -291,6 +291,8 @@ $markdown += New-MDHeader "Xcode" -Level 3
 $markdown += Build-XcodeTable $xcodeInfo | New-MDTable
 $markdown += New-MDNewLine
 
+$markdown += Build-XcodeCommandLineToolsSection
```
Comment (maxim-lobanov) [inline images/macos/software-report/SoftwareReport.Generator.ps1:294]: I suggest to put this information at the begin of documentation rather than Xcode section since CLT is not related to Xcode.
Screen Shot 2020-12-07 at 8 27 13 AM

File: images/macos/software-report/SoftwareReport.Xcode.psm1 [hunk 4]
```diff
@@ -220,6 +220,40 @@ function Build-XcodeSimulatorsTable {
...
+        81 {"8.3"}
+        90 {"9.2"}
+        91 {"9.4"}
+        100 {"10.3"}
+        110 {"11.5"}
+        default   {"12.0"}
+    }
+
+    $versions = @(
+        "clang version $clang_version",
```
Comment (maxim-lobanov) [inline images/macos/software-report/SoftwareReport.Xcode.psm1:247]: I don't think that we need Clang version here since it always equal to CLT version

File: images/macos/software-report/SoftwareReport.Generator.ps1 (hunk 3 above)
Comment (AlenaSviridenko) [reply images/macos/software-report/SoftwareReport.Generator.ps1:294]: Let's put it to the Xcode section but with title "Command Line Tools: <version>"

Comment (maxim-lobanov) [reply images/macos/software-report/SoftwareReport.Generator.ps1:294]: Sounds good. Let's put it to Xcode section before Xcode table

**Review by AlenaSviridenko — APPROVED** [review]

File: images/macos/software-report/SoftwareReport.Xcode.psm1 [hunk 5]
```diff
@@ -220,6 +220,17 @@ function Build-XcodeSimulatorsTable {
     }
 }
 
+function Build-XcodeCommandLineToolsSection {
+    $version = (pkgutil --pkg-info com.apple.pkg.CLTools_Executables | grep "version" | cut -d' ' -f 2 | cut -d'.' -f 1,2)
```
Comment (al-cheb) [inline images/macos/software-report/SoftwareReport.Xcode.psm1:224]: nitpick: unnecessary remove ()

File: images/macos/software-report/SoftwareReport.Xcode.psm1 [hunk 6]
```diff
@@ -220,6 +220,17 @@ function Build-XcodeSimulatorsTable {
...
+
+    $versions = @(
```
Comment (al-cheb) [inline images/macos/software-report/SoftwareReport.Xcode.psm1:226]: nitpick: less an clear
```
New-MDList -Style Unordered -Lines @(
           "Command Line Tools: $version"
)
```

**Review by al-cheb — APPROVED** [review]

**Comment by AlenaSviridenko** [comment]
Current output looks weird, sorry for that idea. Let's put this to `Tools` section where we have fastlane and all that tools.
image
